pkg/icingadb/sla.go

```diff
@@ -0,0 +1,144 @@
+package icingadb
+
+import (
+	"context"
+	"fmt"
+	"github.com/icinga/icingadb/pkg/com"
+	"github.com/icinga/icingadb/pkg/contracts"
+	"github.com/icinga/icingadb/pkg/icingadb/objectpacker"
+	v1 "github.com/icinga/icingadb/pkg/icingadb/v1"
+	"github.com/icinga/icingadb/pkg/types"
+	"github.com/icinga/icingadb/pkg/utils"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+	"time"
+)
+
+type SlaLifecycle struct {
+	v1.EntityWithoutChecksum `json:",inline"`
+	v1.EnvironmentMeta       `json:",inline"`
+	HostId                   types.Binary    `json:"host_id"`
+	ServiceId                types.Binary    `json:"service_id"`
+	CreateTime               types.UnixMilli `json:"create_time"`
+	DeleteTime               types.UnixMilli `json:"delete_time"`
+
+	// The original checkable entity from which this sla lifecycle were transformed
+	OrgEntity contracts.Entity `json:"-" db:"-"`
+}
```

Comment:
I was also thinking the same at the beginning, but then icingadb wouldn't build, since there will be some dependence cycles.

Reply:
What did you put there? Just the struct definition or this whole file?

Reply:
The whole file.

Reply:
Ok, so this should probably be split as well. Most structs that represent database tables are somewhere inside `pkg/icingadb/v1/` whereas the code working with them and writing them to the database is somewhere else.

Reply:
It should probably be in `pkg/icingadb/v1/history/` instead of `pkg/icingadb/v1/`. At least the other types like `SlaHistoryState` and `SlaHistoryDowntime` are defined in `history/` as well. Or would this again result in some import cycles?